### PR TITLE
Added sign in through Facebook (OmniAuth)

### PR DESCRIPTION
**Pull request by dtrypt**
I got Facebook authentication to integrate with devise! Users can login automatically with Facebook now, or they can make an account the normal way. Try it out!
If you guys ever want to try authenticating app this is the resource I used: https://github.com/plataformatec/devise/wiki/OmniAuth:-Overview
And you have to set up some developer account stuff on the provider website. I think Facebook is good for now as far as social media, but I (or someone else) can always add more later on.
To note: - I added two columns to the Users table: name & image
- Added Figaro gem but didn't end up using it. I'll be using it for production though.
- I tried to write tests to validate Omniauth was working properly but failed at that miserably. However, I tested it with fake facebook accounts and it works well. (facebook lets you generate up to 2000 fake accounts in developer tools, pretty cool!)

### CODE REVIEW

**Review by rajastakilos — APPROVED** [review]
Approved.

**Comment by LesleyPark** [comment]
Approved. Nicely done!

**Review by LesleyPark — APPROVED** [review]
Approved. Nicely done!

File: app/models/user.rb [hunk 1]
```diff
@@ -3,5 +3,19 @@ class User < ActiveRecord::Base
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :trackable, :validatable
-end
+         :recoverable, :rememberable, :trackable, :validatable,
+         :omniauthable, :omniauth_providers => [:facebook]
+
+  def self.from_omniauth(auth)
+  where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
```
Comment (sungwon) [inline app/models/user.rb:10]: Just a minor change: indent this so it's under the method and so the contents of the block from line 11 to 17 are indented under where.

**Comment by sungwon** [comment]
Please see Slack for additional tasks that need to be done before (or after) this is merged.

**Pull request merged**